client/my-sites/checkout/upsell-nudge/purchase-modal/index.tsx

```diff
@@ -109,6 +125,44 @@ export default function PurchaseModalWrapper( props: PurchaseModalProps ) {
 		]
 	);
 
+	useEffect( () => {
+		let isUpdatingCart = false;
+
+		if ( storedCard && countries?.length && ! isUpdatingCart ) {
+			const vatDetails: VatDetails = {
+				country: storedCard.tax_location?.country_code,
+				id: storedCard.tax_location?.vat_id,
+				name: storedCard.tax_location?.organization,
+				address: storedCard.tax_location?.address,
+			};
+			const contactInfo: ManagedContactDetails = {
+				state: wrapValueInManagedValue( storedCard.tax_location?.subdivision_code ),
+				city: wrapValueInManagedValue( storedCard.tax_location?.city ),
+				postalCode: wrapValueInManagedValue( storedCard.tax_location?.postal_code ),
+			};
+
+			updateCartContactDetailsForCheckout(
+				countries ?? [],
+				responseCart,
+				updateLocation,
+				contactInfo,
+				vatDetails
+			);
+			replaceProductsInCart( [ productToAdd ] );
+		}
+
+		return () => {
+			isUpdatingCart = true;
+		};
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Minor but could you add a comment here to explain why we're ignoring this and/or which dependencies we want to ignore? That will help anyone in the future who needs to edit this function. (Sometimes I write "// We only want to run this effect when X changes".)